homeassistant/components/nws/weather.py

```diff
@@ -0,0 +1,306 @@
+import logging
+from statistics import mean
+
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components.weather import (
+    WeatherEntity, PLATFORM_SCHEMA, ATTR_FORECAST_CONDITION,
+    ATTR_FORECAST_TEMP, ATTR_FORECAST_TIME,
+    ATTR_FORECAST_WIND_SPEED, ATTR_FORECAST_WIND_BEARING)
+from homeassistant.const import (
+    CONF_API_KEY, CONF_NAME, CONF_LATITUDE, CONF_LONGITUDE, CONF_MODE,
+    LENGTH_KILOMETERS, LENGTH_METERS, LENGTH_MILES, PRESSURE_HPA, PRESSURE_PA,
+    PRESSURE_INHG, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers import config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.util.distance import convert as convert_distance
+from homeassistant.util.pressure import convert as convert_pressure
+from homeassistant.util.temperature import convert as convert_temperature
+
+REQUIREMENTS = ['pynws==0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Data from National Weather Service/NOAA'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+CONF_STATION = 'station'
+
+ATTR_FORECAST_DETAIL_DESCRIPTION = 'detailed_description'
+ATTR_FORECAST_PRECIP_PROB = 'precipitation_probability'
+ATTR_FORECAST_DAYTIME = 'daytime'
+
+# Ordered so that a single condition can be chosen from multiple weather codes.
+# Known NWS conditions that do not map: cold
+CONDITION_CLASSES = OrderedDict([
+    ('snowy', ['snow', 'snow_sleet', 'sleet', 'blizzard']),
+    ('snowy-rainy', ['rain_snow', 'rain_sleet', 'fzra',
+                     'rain_fzra', 'snow_fzra']),
+    ('hail', []),
+    ('lightning-rainy', ['tsra', 'tsra_sct', 'tsra_hi']),
+    ('lightning', []),
+    ('pouring', []),
+    ('rainy', ['rain', 'rain_showers', 'rain_showers_hi']),
+    ('windy-variant', ['wind_bkn', 'wind_ovc']),
+    ('windy', ['wind_skc', 'wind_few', 'wind_sct']),
+    ('fog', ['fog']),
+    ('clear', ['skc']),  # sunny and clear-night
+    ('cloudy', ['bkn', 'ovc']),
+    ('partlycloudy', ['few', 'sct'])
+])
+
+FORECAST_CLASSES = {
+    ATTR_FORECAST_DETAIL_DESCRIPTION: 'detailedForecast',
+    ATTR_FORECAST_TEMP: 'temperature',
+    ATTR_FORECAST_TIME: 'startTime',
+}
+
+FORECAST_MODE = ['daynight', 'hourly']
+
+WIND_DIRECTIONS = ['N', 'NNE', 'NE', 'ENE',
+                   'E', 'ESE', 'SE', 'SSE',
+                   'S', 'SSW', 'SW', 'WSW',
+                   'W', 'WNW', 'NW', 'NNW']
+
+WIND = {name: idx * 360 / 16 for idx, name in enumerate(WIND_DIRECTIONS)}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_MODE, default='daynight'): vol.In(FORECAST_MODE),
+    vol.Optional(CONF_STATION, default=''): cv.string,
+    vol.Required(CONF_API_KEY): cv.string
+})
+
+
+def parse_icon(icon):
+    """
+    Parse icon url to NWS weather codes.
+
+    Example:
```

Comment:
Remove example as it's not working aka gives a HTTP 404 error.

Reply:
The example was indeed wrong, and it is now fixed.  I do think it adds legibility to the code, but let me know if you'd still prefer it to be removed

Reply:
I also included a link to all the icon weather codes above CONDITION_CLASSES, instead of just mentioning only the one code that didn't map.  There are others like `dust`.